app/uk/gov/hmrc/gform/gform/NewFormController.scala

```diff
@@ -212,22 +212,48 @@ class NewFormController(
       implicit request => implicit l => cache =>
         val queryParams: QueryParams = QueryParams.fromRequest(request)
 
-        val formIdData = FormIdData.Plain(UserId(cache.retrievals), formTemplateId)
-        handleForm(formIdData, cache.formTemplate)(newForm(formTemplateId, cache, queryParams)) { form =>
-          cache.formTemplate.draftRetrievalMethod match {
-            case NotPermitted =>
-              fastForwardService.deleteForm(formTemplateId, cache.toAuthCacheWithForm(form, noAccessCode), queryParams)
-            case OnePerUser(ContinueOrDeletePage.Skip) | FormAccessCodeForAgents(ContinueOrDeletePage.Skip) =>
-              auditService.sendFormResumeEvent(form, cache.retrievals)
-              redirectContinue[SectionSelectorType.Normal](cache, form, noAccessCode)
-            case _ =>
-              auditService.sendFormResumeEvent(form, cache.retrievals)
-              val continueFormPage = new ContinueFormPage(cache.formTemplate, choice)
-              Ok(continue_form_page(frontendAppConfig, continueFormPage)).pure[Future]
-          }
-        }
+        for {
+          formTemplate <- getLatestFormTemplate(formTemplateId)
+          formIdData   <- Future.successful(FormIdData.Plain(UserId(cache.retrievals), formTemplate._id))
+          res <-
+            handleForm(formIdData, formTemplate)(
+              newForm(formTemplate._id, cache.copy(formTemplate = formTemplate), queryParams)
+            ) { form =>
+              formTemplate.draftRetrievalMethod match {
+                case NotPermitted =>
+                  fastForwardService.deleteForm(
+                    formTemplate._id,
+                    cache.toAuthCacheWithForm(form, noAccessCode),
+                    queryParams
+                  )
+                case OnePerUser(ContinueOrDeletePage.Skip) | FormAccessCodeForAgents(ContinueOrDeletePage.Skip) =>
+                  auditService.sendFormResumeEvent(form, cache.retrievals)
+                  redirectContinue[SectionSelectorType.Normal](
+                    cache.copy(formTemplate = formTemplate),
+                    form,
+                    noAccessCode
+                  )
+                case _ =>
+                  auditService.sendFormResumeEvent(form, cache.retrievals)
+                  val continueFormPage = new ContinueFormPage(formTemplate, choice)
+                  Ok(continue_form_page(frontendAppConfig, continueFormPage)).pure[Future]
+              }
+            }
+        } yield res
     }
 
+  private def getLatestFormTemplate(
+    formTemplateId: FormTemplateId
+  )(implicit hc: HeaderCarrier): Future[FormTemplate] =
+    for {
+      formTemplate <- gformConnector.getFormTemplate(formTemplateId)
+      res <-
+        formTemplate.redirect match {
+          case Some(ft) => getLatestFormTemplate(ft)
+          case None     => Future.successful(formTemplate.formTemplate)
```

Comment:
Can we do this with just one call to the backend?

We should add new method to the `gformConnector`: 
```
def getLatestFormTemplate(
  formTemplateId: FormTemplateId
)
```

And we should implement it on the backend in terms of `formRedirect` collection.

We have all the data ready for us to be processed.
```
> db.formRedirect.find({}).pretty();
{
	"_id" : "version-test-1",
	"redirect" : "version-test",
	"createdAt" : "2022-02-14T13:59:00.232754Z",
	"updatedAt" : "2022-02-14T13:59:00.232756Z"
}
{
	"_id" : "version-test-0",
	"redirect" : "version-test-1",
	"createdAt" : "2022-02-14T13:59:39.306877Z",
	"updatedAt" : "2022-02-14T13:59:39.306879Z"
}
```